src/components/VideoCapture/index.tsx

```diff
@@ -0,0 +1,231 @@
+
+const initialStateWithoutMediaStream: Omit<State, 'hasMediaStream'> = {
+  hasBecomeInactive: false,
+  hasCameraError: false,
+  hasRecordingTakenTooLong: false,
+  isRecording: false,
+}
+
+const recordingTooLongError: ErrorProp = {
+  name: 'FACE_VIDEO_TIMEOUT',
+  type: 'warning',
+}
+
+export default class VideoCapture extends Component<Props, State> {
+  private webcam?: Webcam
+
+  state = { ...initialStateWithoutMediaStream, hasMediaStream: false }
+
+  startRecording = (): void => {
+    this.webcam && this.webcam.startRecording()
+    this.setState({ isRecording: true, hasBecomeInactive: false })
+  }
+
+  stopRecording = (): void => {
+    this.webcam && this.webcam.stopRecording()
+    this.setState({ isRecording: false })
+  }
+
+  handleRecordingStart = (): void => {
+    if (this.state.hasMediaStream) {
+      this.startRecording()
+      this.props.onRecordingStart && this.props.onRecordingStart()
+    }
+  }
+
+  handleRecordingStop = (): void => {
+    const { hasRecordingTakenTooLong } = this.state
+    this.stopRecording()
+
+    if (this.webcam && !hasRecordingTakenTooLong) {
+      getRecordedVideo(this.webcam, (payload) =>
+        this.props.onVideoCapture(payload)
+      )
+    }
+  }
+
+  handleMediaStream = (): void => this.setState({ hasMediaStream: true })
+
+  handleInactivityTimeout = (): void =>
+    this.setState({ hasBecomeInactive: true })
+
+  handleRecordingTimeout = (): void => {
+    this.setState({ hasRecordingTakenTooLong: true })
+    this.stopRecording()
+  }
+
+  handleCameraError = (): void => this.setState({ hasCameraError: true })
+
+  handleFallbackClick = (callback?: () => void): void => {
+    this.setState({ ...initialStateWithoutMediaStream }, () => {
+      this.props.onRedo()
+      typeof callback === 'function' && callback()
+    })
+  }
+
+  renderRedoActionsFallback: RenderFallbackProp = (text, callback) => (
+    <FallbackButton
+      text={text}
+      onClick={() => this.handleFallbackClick(callback)}
+    />
+  )
+
+  renderError = (): h.JSX.Element => {
+    const { inactiveError, renderFallback, trackScreen } = this.props
+    const passedProps = this.state.hasRecordingTakenTooLong
+      ? {
+          error: recordingTooLongError,
+          hasBackdrop: true,
+          renderFallback: this.renderRedoActionsFallback,
+        }
+      : {
+          error: inactiveError,
+          isDismissible: true,
+          renderFallback,
+        }
+
+    return <CameraError trackScreen={trackScreen} {...passedProps} />
+  }
+
+  renderInactivityTimeoutMessage = (): h.JSX.Element | null => {
+    const {
+      hasBecomeInactive,
+      hasCameraError,
+      hasRecordingTakenTooLong,
+      isRecording,
+    } = this.state
+    const hasError =
+      hasRecordingTakenTooLong || hasCameraError || hasBecomeInactive
+
+    if (hasError) {
+      return null
+    }
+
+    const passedProps = {
+      key: isRecording ? 'recording' : 'notRecording',
+      seconds: isRecording ? 20 : 12,
```

Comment:
Might be worth assigning these values to a constant

Reply:
Yes in the next PR, those values were checked-out as constants: https://github.com/onfido/onfido-sdk-ui/blob/feature/document-video-capture/src/components/utils/constants.ts